### DISCUSSION TITLE
Make the expanded rows manageable from outside like selectedRecords

### DISCUSSION THREAD

**Discussion by asa75** [icflorescu/mantine-datatable#72] [Ideas]
Can you add the possibility to control which rows are expanded like for initiallyExpanded but at the runtime as example like with  selectedRecords and onSelectedRecordsChange props? So we can customize with actions cell the open/collapse rows.

**Comment by icflorescu** [comment]
Sounds useful. Will do soon.
Added referring issue: https://github.com/icflorescu/mantine-datatable/issues/73.

**Comment by icflorescu** [comment]
Done in `v1.7.9`. Can be used like this:

```ts
<DataTable
  // ...
  rowExpansion={{
    // ...
    expanded: {
      recordIds: expandedRecordIds,
      onRecordIdsChange: setExpandedRecordIds,
    },
  }}
/>
```

Full working example in **Controlled mode** section of [this page](https://icflorescu.github.io/mantine-datatable/examples/expanding-rows).
